### PR TITLE
Wrote the necessary tests, but unfortunately, as correct as it seems,…

### PR DESCRIPTION
**Pull request by 0xdopewilly**
… ERROR

### CODE REVIEW

File: test/Attendance.t.sol [hunk 1]
```diff
@@ -135,32 +135,95 @@ contract AttendanceTest is Test {
...
+        age: 18,
+        attendanceCount: 0,
+        name: student0Name,
+        isRegistered: false,
+        studentAddress: student0Addr
+    });
+
+    // Attempt to create a student using an unauthorized account
+    vm.prank(unauthorizedAcct);
+    attendance.createStudent(student_);
```
Comment (ikwuoz) [inline test/Attendance.t.sol]: add assert to check test is correct.

File: test/Attendance.t.sol [hunk 2]
```diff
@@ -135,32 +135,95 @@ contract AttendanceTest is Test {
...
+
 
     // test authorized reps can increment attendance score
-    function test_CourseRepsCanIncrmentAttendance() public pure {
-        assert(true);
-    }
+    function test_CourseRepsCanIncrmentAttendance() public {
+    // Authorize a course rep first
+    vm.prank(lecturer);
+    address;
```
Comment (ikwuoz) [inline test/Attendance.t.sol]: whats the purpose of this line?

File: test/Attendance.t.sol [hunk 3]
```diff
@@ -135,32 +135,95 @@ contract AttendanceTest is Test {
...
-        assert(false);
-    }
+    function testFail_UnauthorizedAcctsCannotIncrementAttendance() public {
+    // Register a student first
+    vm.prank(lecturer);
+    attendance.registerStudentForCourse(CSC201, student0Addr);
+
+    // Attempt to increment attendance using an unauthorized account
+    vm.prank(unauthorizedAcct);
+    attendance.incrementAttendance(CSC201, student0Addr);
```
Comment (ikwuoz) [inline test/Attendance.t.sol]: add assert(s) to check test is correct.

**Review by ikwuoz — COMMENTED** [review]
add the changes suggested and i would approve.

File: test/Attendance.t.sol [hunk 4]
```diff
@@ -118,7 +117,7 @@ contract AttendanceTest is Test {
             studentAddress: student2Addr
         });
 
-        Student[] memory students_ = new Student[](3);
+        Student;
```
Comment (ikwuoz) [inline test/Attendance.t.sol:119]: this the issue, please fix.

File: test/Attendance.t.sol (hunk 2 above)
Comment (0xdopewilly) [reply test/Attendance.t.sol]: > whats the purpose of this line?
The line address; within the function test_CourseRepsCanIncrmentAttendance() seems to be mistakenly placed. Didnt notice it imagine, so as it is now, its performing no function. I will refactor the code and push a commit.

**Review by ikwuoz — APPROVED** [review]
i'll merge and refactor.

**Review by ikwuoz — APPROVED** [review]

**Pull request merged**